pkg/beholder/message.go

```diff
@@ -0,0 +1,257 @@
+	Attrs Attributes
+	Body  []byte
+}
+
+type Metadata struct {
+	//	REQUIRED FIELDS
+	// Schema Registry URI to fetch schema
+	BeholderDataSchema string `validate:"required,uri"`
+
+	// OPTIONAL FIELDS
+	// The version of the CL node.
+	NodeVersion string
+	// mTLS public key for the node operator. This is used as an identity key but with the added benefit of being able to provide signatures.
+	NodeCsaKey string
+	// Signature from CSA private key.
+	NodeCsaSignature string
+	DonID            string
+	// The RDD network name the CL node is operating with.
+	NetworkName          []string
+	WorkflowID           string
+	WorkflowName         string
+	WorkflowOwnerAddress string
+	// Hash of the workflow spec.
+	WorkflowSpecID string
+	// The unique execution of a workflow.
+	WorkflowExecutionID string
+	// The address for the contract.
+	CapabilityContractAddress string
+	CapabilityID              string
+	CapabilityVersion         string
+	CapabilityName            string
+	NetworkChainID            string
+}
+
+func (m Metadata) Attributes() Attributes {
+	return Attributes{
+		"node_version":                m.NodeVersion,
+		"node_csa_key":                m.NodeCsaKey,
+		"node_csa_signature":          m.NodeCsaSignature,
+		"don_id":                      m.DonID,
+		"network_name":                m.NetworkName,
+		"workflow_id":                 m.WorkflowID,
+		"workflow_name":               m.WorkflowName,
+		"workflow_owner_address":      m.WorkflowOwnerAddress,
+		"workflow_spec_id":            m.WorkflowSpecID,
+		"workflow_execution_id":       m.WorkflowExecutionID,
+		"beholder_data_schema":        m.BeholderDataSchema,
+		"capability_contract_address": m.CapabilityContractAddress,
+		"capability_id":               m.CapabilityID,
+		"capability_version":          m.CapabilityVersion,
+		"capability_name":             m.CapabilityName,
+		"network_chain_id":            m.NetworkChainID,
+	}
+}
+
+type Attributes map[string]any
+
+func NewAttributes(attrKVs ...any) Attributes {
+	attrs := make(Attributes, len(attrKVs)/2)
+	attrs.Add(attrKVs...)
+	return attrs
+}
+
+func (a Attributes) Add(attrKVs ...any) Attributes {
+	l := len(attrKVs)
+	for i := 0; i < l; {
+		switch t := attrKVs[i].(type) {
+		case map[string]any:
+			for k, v := range t {
+				a[k] = v
+			}
+			i++
+		case Attributes:
+			for k, v := range t {
+				a[k] = v
+			}
+			i++
+		case string:
+			if i+1 >= l {
+				break
+			}
+			val := attrKVs[i+1]
+			a[t] = val
+			i += 2
+		default:
+			// Unexpected type
+			return a
+		}
+	}
+	return a
+}
+
+func NewMessage(body []byte, attrKVs ...any) Message {
+	return Message{
+		Body:  body,
+		Attrs: NewAttributes(attrKVs...),
+	}
+}
+
+func (e *Message) AddAttributes(attrKVs ...any) {
+	attrs := NewAttributes(attrKVs...)
+	if e.Attrs == nil {
+		e.Attrs = make(map[string]any, len(attrs))
+	}
+	for k, v := range attrs {
+		e.Attrs[k] = v
+	}
+}
+
+func (e *Message) AddOtelAttributes(attrs ...attribute.KeyValue) {
```

Comment:
Is it going to be clear to the consumer when they should use `AddAttributes` vs `AddOtelAttributes`?  You've put in a lot of work to add functionality around a `map[string]any`, so you could leverage the existing `AddAttributes` and add this otel case to the type switch case statement in `Add`.